### PR TITLE
Local changes

### PR DESCRIPTION
**Pull request by joaobrittoneto**
@doudou,
I'm testing and debugging the code and found a problem.
The task goes to exception state because of timeout in readPacket. I think it's a common case once the undewater connection is not reliable and the auv_uwmodem_data can receive corrupted/garbage stream of data at any time.  
I'm fixing it by using a try/catch in the [library](https://github.com/Brazilian-Institute-of-Robotics/drivers-auv_uwmodem_data/pull/4), but now it does not clear the extractPakcet buffer. Do you know how i can do it? 
Is it the best solution?

### CODE REVIEW

**Comment by doudou** [comment]
Recovering from a timeout makes sense. But I would definitely do this at the component level, removing the horrible special-packet-type you had to add at the driver level.

As for clearing the driver, there is clear(), but it does not empty the internal buffer :( I can do it there if you need it.

**Comment by joaobrittoneto** [comment]
Unit test runs without error, but depends on https://github.com/rock-drivers/drivers-iodrivers_base/pull/9.
@doudou, in the unit test, I tried to connect [ports](https://github.com/Brazilian-Institute-of-Robotics/drivers-orogen-auv_uwmodem_data/blob/local_changes/test/test_base.rb#L112) but it didn't really work. I had to write data direct in the ports. Is that right?

**Comment by doudou** [comment]
Depends on what you tried to connect. You can connect ports to ports just fine. However, the data accessors created by the reader and writer declarations are data reader/writer, not ports, so those you can't connect.

**Comment by joaobrittoneto** [comment]
ah ok, understood

File: tasks/Task.cpp [hunk 1]
```diff
@@ -65,12 +68,12 @@ void Task::updateHook()
 
     // Put raw_data_input in queue to be delivered.
     iodrivers_base::RawPacket raw_data_input;
-    while(_raw_data_input.read(raw_data_input) == RTT::NewData)
-        sendOnePacket(fillPacket(dataParser.marshalText(raw_data_input), TEXT));
+    if(_raw_data_input.read(raw_data_input) == RTT::NewData)
```
Comment (doudou) [inline tasks/Task.cpp]: why the change from while to if ? This is most likely to be wrong !

File: tasks/UWModemDataAUV.cpp [hunk 2]
```diff
@@ -42,26 +42,26 @@ void UWModemDataAUV::updateHook()
 
     // Put job_notification in queue to be delivered.
     JobNotification job_notification;
-    while(_job_notification.read(job_notification) == RTT::NewData) {
```
Comment (doudou) [inline tasks/UWModemDataAUV.cpp]: Same here, why ?????

**Comment by doudou** [comment]
I pushed the change to clear().

Also, can you explain the change from while to if ? It makes no sense at all from my point of view.

Finally, would you mind cleaning this branch before we merge ?

File: tasks/Task.cpp (hunk 1 above)
Comment (joaobrittoneto) [inline tasks/Task.cpp]: No major reason. I was debugging and though it could be a possible reason of error, been stucked in a while loop. I'll set them back to while.

**Comment by joaobrittoneto** [comment]
What do you mean by cleaning this branch?

**Comment by doudou** [comment]
Each commit is basically a mix of stuff, some are or will be reverted later in the branch (e.g. the while<>if change, ...). "local changes on ff-nav" is really not properly explaining anything. That kind of stuff.

Honestly the best option would IMO be to run `git reset autobuild/master` to leave the working copy untouched but remove the commits, and then re-commit it piece-by-piece by using `git gui`

**Comment by joaobrittoneto** [comment]
I think it's quite ok now.

**Comment by doudou** [comment]
Much better. Far from perfect, but that'll do. Feel free to merge.

**Comment by joaobrittoneto** [comment]
Merging
